File: is-subsequence/is_subsequence.go
func isSubsequence(s string, t string) bool {
    var i, j int
    for i < len(s) && j < len(t) {
        if s[i] == t[j] {
            i += 1
        }
        j += 1
    }
    return i == len(s)
}

File: is-subsequence/is_subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        index = 0
        for i in range(len(s)):
            index = t.find(s[i], index)
            if index < 0:
                return False
            index += 1
        return True